src/js/utils/JSONUtil.js

```diff
@@ -155,23 +154,31 @@ module.exports = {
         case 'begin-object':
           // Keep track of keyed objects
           if (objectKey) {
-            keyLines.push({
+            let blockToken = {
```

Comment:
I see a repeating pattern here. Massive Switch/case is a different story, but maybe we could think a way to generalize this token creation, how do you think?